File: app/src/main/java/com/stedi/randomimagegenerator/app/view/fragments/base/ButterKnifeFragment.java
package com.stedi.randomimagegenerator.app.view.fragments.base;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.View;

import butterknife.ButterKnife;
import butterknife.Unbinder;

public abstract class ButterKnifeFragment extends BaseFragment {
    private Unbinder unbinder;

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        unbinder = ButterKnife.bind(this, view);
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        unbinder.unbind();
    }
}

File: app/src/main/java/com/stedi/randomimagegenerator/app/view/fragments/base/ButterKnifeFragment.kt
package com.stedi.randomimagegenerator.app.view.fragments.base

import android.os.Bundle
import android.view.View
import butterknife.ButterKnife
import butterknife.Unbinder

abstract class ButterKnifeFragment : BaseFragment() {
    private var unbinder: Unbinder? = null

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        unbinder = ButterKnife.bind(this, view)
    }

    override fun onDestroyView() {
        super.onDestroyView()
        unbinder?.unbind()
    }
}